core/trino-main/src/main/java/io/trino/execution/scheduler/BroadcastOutputBufferManager.java

```diff
@@ -19,57 +19,46 @@
 import javax.annotation.concurrent.GuardedBy;
 import javax.annotation.concurrent.ThreadSafe;
 
-import java.util.List;
-import java.util.function.Consumer;
-
 import static io.trino.execution.buffer.OutputBuffers.BROADCAST_PARTITION_ID;
 import static io.trino.execution.buffer.OutputBuffers.BufferType.BROADCAST;
 import static io.trino.execution.buffer.OutputBuffers.createInitialEmptyOutputBuffers;
-import static java.util.Objects.requireNonNull;
 
 @ThreadSafe
 class BroadcastOutputBufferManager
         implements OutputBufferManager
 {
-    private final Consumer<OutputBuffers> outputBufferTarget;
-
     @GuardedBy("this")
     private OutputBuffers outputBuffers = createInitialEmptyOutputBuffers(BROADCAST);
 
-    public BroadcastOutputBufferManager(Consumer<OutputBuffers> outputBufferTarget)
-    {
-        this.outputBufferTarget = requireNonNull(outputBufferTarget, "outputBufferTarget is null");
-        outputBufferTarget.accept(outputBuffers);
-    }
-
     @Override
-    public void addOutputBuffers(List<OutputBufferId> newBuffers, boolean noMoreBuffers)
```

Comment:
nit: I guess the refactoring commit could be split to make reviewing easier. E.g changes to `BroadcastOutputBufferManager` could be separate commit I think.